cobra/flux_analysis/phenotype_phase_plane.py

```diff
@@ -374,168 +372,201 @@ def production_envelope(model, reactions, objective=None, c_source=None,
+                        continue
+                    if abs(model.solver.objective.value) < \
+                            model.solver.configuration.tolerances.feasibility:
+                        grid.at[i, 'flux_{}'.format(direction)] = 0.0
+                    else:
+                        grid.at[i, 'flux_{}'.format(direction)] = \
+                            model.solver.objective.value
+                    if c_input is not None:
+                        grid.at[i, 'carbon_yield_{}'.format(direction)] = \
+                            total_yield([rxn.flux for rxn in c_input],
+                                        input_components,
+                                        model.solver.objective.value,
+                                        output_components)
+                        grid.at[i, 'mass_yield_{}'.format(direction)] = \
+                            total_yield([rxn.flux for rxn in c_input],
+                                        input_weights,
+                                        model.solver.objective.value,
+                                        output_weight)
+
+
+def total_yield(input_fluxes, input_elements, output_flux, output_elements):
+    """
+    Compute total output per input unit.
 
+    Units are typically mol carbon atoms or gram of source and product.
 
-def carbon_yield(c_input_output):
-    """ mol product per mol carbon input
+    Parameters
+    ----------
+    input_fluxes : list
+        A list of input reaction fluxes in the same order as the
+        ``input_components``.
+    input_elements : list
+        A list of reaction components which are in turn list of numbers.
+    output_flux : float
+        The output flux value.
+    output_elements : list
+        A list of stoichiometrically weighted output reaction components.
 
     Returns
     -------
     float
-        the mol carbon atoms in the product (as defined by the model
-        objective) divided by the mol carbon in the input reactions (as
-        defined by the model medium) or zero in case of division by zero
-        arises
+        The ratio between output (mol carbon atoms or grams of product) and
+        input (mol carbon atoms or grams of source compounds).
     """
 
-    c_input, c_output = c_input_output
-    if c_input is None:
-        return nan
-    carbon_input_flux = total_carbon_flux(c_input, consumption=True)
-    carbon_output_flux = total_carbon_flux(c_output, consumption=False)
+    carbon_input_flux = sum(
+        total_components_flux(flux, components, consumption=True)
+        for flux, components in zip(input_fluxes, input_elements))
+    carbon_output_flux = total_components_flux(
+        output_flux, output_elements, consumption=False)
     try:
         return carbon_output_flux / carbon_input_flux
     except ZeroDivisionError:
         return nan
 
 
-def mass_yield(c_input_output):
-    """Gram product divided by gram of carbon input source
+def reaction_components(reaction, atom='C'):
```

Comment:
`element` not atom.. And since you add this, perhaps do it all the way and allow user to calculate any element yield by having e.g. `yield_element='N'` as argument to `production_envelope`? Or do later and remove parametrisation here..

Reply:
Removing parametrization here for now. It's a little annoying with column naming so unless there is a need for it I'd skip it.